src/agentscope/model/_zhipu_model.py

```diff
@@ -0,0 +1,394 @@
+            start_datetime (`datetime`):
+                The start datetime of the response generation.
+            response (`Any`):
+                Zhipu AI streaming response to parse.
+            structured_model (`Type[BaseModel] | None`, default `None`):
+                A Pydantic BaseModel class that defines the expected structure
+                for the model's output.
+
+        Returns:
+            AsyncGenerator[ChatResponse, None]:
+                An async generator that yields ChatResponse objects containing
+                the content blocks and usage information for each chunk in the
+                streaming response.
+        """
+        accumulated_text = ""
+        accumulated_thinking = ""
+        tool_calls = OrderedDict()  # Store tool calls
+        metadata = None
+
+        for chunk in response:
+            if not chunk.choices:
+                continue
+
+            choice = chunk.choices[0]
+            delta = choice.delta
+
+            # Handle text content
+            if delta.content:
+                accumulated_text += delta.content
+
+            # Handle thinking/reasoning content
+            if hasattr(delta, "reasoning_content") and delta.reasoning_content:
+                accumulated_thinking += str(delta.reasoning_content)
+
+            # Handle tool calls
+            if delta.tool_calls:
+                for tool_call in delta.tool_calls:
+                    tool_id = tool_call.id or f"call_{tool_call.index}"
+                    if tool_id not in tool_calls:
+                        tool_calls[tool_id] = {
+                            "type": "tool_use",
+                            "id": tool_id,
+                            "name": tool_call.function.name or "",
+                            "input": "",
+                        }
+
+                    if tool_call.function.name:
+                        tool_calls[tool_id]["name"] = tool_call.function.name
+                    if tool_call.function.arguments:
+                        tool_calls[tool_id][
+                            "input"
+                        ] += tool_call.function.arguments
+
+            # Calculate usage statistics
+            current_time = (datetime.now() - start_datetime).total_seconds()
+            usage = None
+            if hasattr(chunk, "usage") and chunk.usage:
+                usage = ChatUsage(
+                    input_tokens=chunk.usage.prompt_tokens or 0,
+                    output_tokens=chunk.usage.completion_tokens or 0,
+                    time=current_time,
+                )
+
+            # Create content blocks
+            contents: list = []
+
+            # Add thinking block if present
+            if accumulated_thinking:
+                contents.append(
+                    ThinkingBlock(
+                        type="thinking",
+                        thinking=accumulated_thinking,
+                    ),
+                )
+
+            if accumulated_text:
```

Comment:
I've discovered an odd issue: when calling glm4.5, in cases where reasoning_content is present, the content is just "\n", which breaks the entire output/display. Maybe we can use `if accumulated_text.strip()` here to handle this edge case.